tests/test_lang/test_tfhub_lang.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from whatlies.language import TFHubLanguage
+
+
+@pytest.fixture
+def lang(request):
+    return TFHubLanguage(**request.param)
+
+
+@pytest.mark.parametrize(
+    "lang, expected_shape",
+    [
+        (
+            {"url": "https://tfhub.dev/google/universal-sentence-encoder-large/5"},
```

Comment:
I like the changes, but I'm unsure if this is an appropriate method of testing. Every time somebody might push a branch we'll download 500+ MB payload per model. 

Instead, is there a super small model that we might download? I'm also wondering if it makes sense to rely on `tf-hub` during testing. If the service ever goes down temporarily it would mess up our tests. Can we download a small model ahead of time instead? That's what I'm doing for `FastText` and `spaCy`.

Reply:
I should acknowledge here that my initial `ConveRTLanguage` also suffers from this phenomenon.

Reply:
> Every time somebody might push a branch we'll download a 500+ MB payload per model.

There are two different aspects to this:
 1. It might be time-consuming to download a large file, so running tests would take longer: one workaround for the this issue for local dev and testing is to mark these tests as slow in pytest and then create a separate command which only runs non-slow tests. In this case, the developer can decide which one to run based on their modifications. Although, this does not work for Github actions, especially if it charges per minute.
 2. It might be a data transfer/storage limit in Github actions, or even on the developers side in their local dev environment: I don't have a solution for this issue!

> I'm unsure if this is an appropriate method of testing

I think some repos like [keras-applications](https://github.com/keras-team/keras-applications) (which basically consists of more than 10 models) use this approach. A single round of test takes around 10 minutes on Travis CI for keras-applications repo.

> is there a super small model that we might download?

Yes, there is. The swivel-20dim model which I have included in the tests is less than 2 MB. If you think we should not use large model files, we can only use that model instead.

> I'm also wondering if it makes sense to rely on tf-hub during testing...

If you ask me, I think it does make sense. Mainly because it's very rare that TF-HUB would be down (after all, it's maintained by Google!), and also I think the users would mostly rely on TF-HUB directly to use this class; so it makes sense to use TF-HUB directly in our tests. Although, even here we can instead download a small model (e.g. swivel-20dim) and put it in the test directory (however, I am not in favor of this approach entirely, even for fasttext or spacy model files as you have done before. That's because the dependencies are also changing and their format of saving/loading might change so those files go stale, which is the case for the current spacy model files in repo; and besides that even if we want to build/save/load a **minimal model** we should do it entirely in the tests to make sure that the library works with the most stable current version of dependencies).

**All in all, let me know of your final opinion on this and I'll make changes accordingly.**

Reply:
Nice! I think your `swivel-20dim` suggestion is a good option. It'd allow us to fully test `tf-hub` as a backend which also means that we can do much lighter testing for items that depend on it like `ConveRT` and `universal-sentence-encoder`. 

*Anekdote* 

TF-HUB shouldn't go down often, that's true ... but I've had this one day where `s3` went down on amazon and because our CI depended on a project that had data hosted there (I think it was keras) our CI also broke because of it. Despite another cloud provider going down, we suddenly couldn't deploy anymore.

Reply:
All right, I just removed the large model files from the tests and used the two small variants of `swivel` model instead. Each has a size of ~1.5 MB.

> Anekdote...

I see! Thanks for sharing your experience.